contracts/StorageManager.sol

```diff
@@ -242,56 +292,113 @@ contract StorageManager {
     @dev
         - if amount is zero then all withdrawable funds are transferred (eq. all available funds minus funds for still non-payed out periods and current period)
         - if Agreement is terminated Consumer can withdraw all remaining funds
+        - if the token is not the native currency, then the contract must be first be given allowance to transfer tokens in it's posession.
     @param dataReference the data reference of agreement to be funds withdrawn from
     @param provider the address of the provider of the Offer.
+    @param tokens the tokens in which to withdraw. By convention, address(0) is the native currency.
+    @param amounts the value you want to withdraw for each token
     */
-    function withdrawFunds(bytes32[] memory dataReference, address provider, uint256 amount) public payable {
+    function withdrawFunds(
+        bytes32[] memory dataReference,
+        address provider,
+        address[] memory tokens,
+        uint256[] memory amounts
+    ) public whenNotPaused {
```

Comment:
we should allow to withdraw even if its paused.

Reply:
Done